### PR TITLE
[PBW-4300] changed button elements to div elements,also created separated event …

### PR DESCRIPTION
**Pull request by shruthi17**
…listeners for all controlbar elements

### CODE REVIEW

**Comment by mistermendez** [comment]
Try changing `<button>` to an `<a>` tag.

Button is an inline element but we are using it like a block element and this may cause inconsistency among browsers.

The`<div>` works because it's a block element but it's not clickable.

The`<a>` tag is a block element and clickable.

File: js/components/controlBar.js [hunk 1]
```diff
@@ -104,8 +104,9 @@ var ControlBar = React.createClass({
   },
 
   toggleQualityPopover: function() {
+    this.props.controller.toggleVideoQualityPopOver();
     this.setState({
-      showVideoQualityPopover: !this.state.showVideoQualityPopover
+      showVideoQualityPopover: this.props.controller.state.videoQualityOptions.showVideoQualityPopover
```
Comment (mistermendez) [inline js/components/controlBar.js]: Why did we change Video Quality Popover logic?

File: js/components/controlBar.js [hunk 2]
```diff
@@ -148,6 +151,62 @@ var ControlBar = React.createClass({
     this.removeHighlight({target: ReactDOM.findDOMNode(this.refs.volumeIcon)});
   },
 
+  playPauseHighlight:function() {
```
Comment (mistermendez) [inline js/components/controlBar.js]: Do we need separate functions for the same logic?

File: js/components/controlBar.js [hunk 3]
```diff
@@ -231,20 +290,20 @@ var ControlBar = React.createClass({
         "oo-live-nonclickable": isLiveNow
       });
 
-    var videoQualityPopover = this.state.showVideoQualityPopover ? <VideoQualityPopover {...this.props} togglePopoverAction={this.toggleQualityPopover}/> : null;
+    var videoQualityPopover = this.props.controller.state.videoQualityOptions.showVideoQualityPopover ? <VideoQualityPopover {...this.props} togglePopoverAction={this.toggleQualityPopover}/> : null;
```
Comment (mistermendez) [inline js/components/controlBar.js]: Why did this change?

File: js/components/controlBar.js [hunk 4]
```diff
@@ -231,20 +290,20 @@ var ControlBar = React.createClass({
...
+      "oo-selected": this.props.controller.state.videoQualityOptions.showVideoQualityPopover
     });
 
     var controlItemTemplates = {
-      "playPause": <button className="oo-play-pause oo-control-bar-item" onClick={this.handlePlayClick} key="playPause">
-        <Icon {...this.props} icon={playIcon}
-          style={dynamicStyles.iconCharacter}
-          onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
-      </button>,
+      "playPause": <div className="oo-play-pause oo-control-bar-item" onClick={this.handlePlayClick} key="playPause">
```
Comment (mistermendez) [inline js/components/controlBar.js]: Elements with `onClick` event handlers need to be clickable elements (i.e.: `<a>` `<button>`) to work consistently on desktop and mobile browsers.

**Comment by mistermendez** [comment]
This update has the following issues:

- Video Quality Popover is broken (be sure to test with an asset that shows all control bar buttons)
- More Options button style is broken
- Discover button does not highlight
- The control bar unit test is broken: FAIL  tests/components/controlBar-test.js
(Run `npm test`. Also add `npm test` to `/html5-skin/.git/hooks/pre-push` to run unit test before pushing to repo)

File: js/components/controlBar.js (hunk 4 above)
Comment (shruthi17) [inline js/components/controlBar.js]: ok,will do it by changing <div> to <a>

**Comment by mistermendez** [comment]
Overall I recommend changing the approach to this issue by simply changing the `<button>` tag to an `<a>` tag, along with corresponding style.

And not modify any logic.

**Comment by shruthi17** [comment]
- All the controlBar Buttons are working fine on MAC Safari, Chrome and Firefox.

- ControlBar is not broken, if we play any asset.
(need to take the local build of skin.json if your on master as it was changed to 90 by default, if it takes default:60 it brokes).

- Agreed, npm test failed :

 FAIL  tests/components/controlBar-test.js (5.001s)
Warning: flattenChildren(...): Encountered two children with the same key, `.$playPause`. Child keys must be unique; when two children share a key, only the first child will be used.
Warning: flattenChildren(...): Encountered two children with the same key, `.$playPause`. Child keys must be unique; when two children share a key, only the first child will be used.
● ControlBar › it creates a control bar
  - TypeError: Cannot read property 'showVideoQualityPopover' of undefined

- will work on this again, Thanks for the update.

**Comment by shruthi17** [comment]
sure, will make it now and keep you posted.

Let me show you my local once, then ill follow your directions.

Thanks

On Tue, Jul 12, 2016 at 1:11 PM, Jeoffrey Mendez <notifications@github.com>
wrote:

> Overall I recommend changing the approach to this issue by simply changing
> the <button> tag to an <a> tag, along with corresponding style.
>
> And not modify any logic.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ooyala/html5-skin/pull/591#issuecomment-232164968>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS2kK8Fr4KeB8YXT4lXNOXthwZI-ve98ks5qU_TvgaJpZM4JKsLf>
> .
>



-- 
Thanks & Regards
Shruthi Sunkara
628-333-4576
https://www.linkedin.com/in/shruthi-sunkara-1b2930108

**Comment by mistermendez** [comment]
Also change buttons in - `/html5-skin/js/components/moreOptionsPanel.js`

**Comment by shruthi17** [comment]
I did already, about to commit the changes


Thanks

On Tue, Jul 12, 2016 at 4:11 PM, Jeoffrey Mendez <notifications@github.com>
wrote:

> Also change buttons in - /html5-skin/js/components/moreOptionsPanel.js
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ooyala/html5-skin/pull/591#issuecomment-232209472>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS2kK7Mcp0R-Ke3VKGbuCij1djAumh3hks5qVB8_gaJpZM4JKsLf>
> .
>



-- 
Thanks & Regards
Shruthi Sunkara
628-333-4576
https://www.linkedin.com/in/shruthi-sunkara-1b2930108

File: gulpfile.js [hunk 5]
```diff
@@ -30,8 +30,8 @@ var path = {
 };
 
 var devServer = {
-  host: '0.0.0.0',
-  port: 4444,
+  host: 'localhost',
```
Comment (mistermendez) [inline gulpfile.js]: Do not change this.

Feel free to update locally but do not commit this change.

File: gulpfile.js [hunk 6]
```diff
@@ -30,8 +30,8 @@ var path = {
...
+  port: 8000,
```
Comment (mistermendez) [inline gulpfile.js]: same here

File: js/components/controlBar.js [hunk 7]
```diff
@@ -261,51 +261,51 @@ var ControlBar = React.createClass({
...
         <Icon {...this.props} icon="ellipsis" style={dynamicStyles.iconCharacter}
           onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
-      </button>,
+      </a>,
 
       "quality": (
-        <div className="oo-popover-button-container" key="quality">
+        <a className="oo-popover-button-container" key="quality">
           {videoQualityPopover}
           <button className={qualityClass} onClick={this.handleQualityClick}>
```
Comment (mistermendez) [inline js/components/controlBar.js]: this should be `<a>`

File: js/components/controlBar.js [hunk 8]
```diff
@@ -261,51 +261,51 @@ var ControlBar = React.createClass({
...
+      "moreOptions": <a className="oo-more-options oo-control-bar-item"
         onClick={this.handleMoreOptionsClick} key="moreOptions">
         <Icon {...this.props} icon="ellipsis" style={dynamicStyles.iconCharacter}
           onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
-      </button>,
+      </a>,
 
       "quality": (
-        <div className="oo-popover-button-container" key="quality">
+        <a className="oo-popover-button-container" key="quality">
```
Comment (mistermendez) [inline js/components/controlBar.js]: this should remain a `<div>`

File: js/components/controlBar.js [hunk 9]
```diff
@@ -240,11 +240,11 @@ var ControlBar = React.createClass({
...
     var controlItemTemplates = {
-      "playPause": <button className="oo-play-pause oo-control-bar-item" onClick={this.handlePlayClick} key="playPause">
+      "playPause": <a className="oo-play-pause oo-control-bar-item" onClick={this.handlePlayClick} key="playPause">
         <Icon {...this.props} icon={playIcon}
           style={dynamicStyles.iconCharacter}
           onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
-      </button>,
+      </a>,
 
       "live": <button className={liveClass}
```
Comment (mistermendez) [inline js/components/controlBar.js]: this should be `<a>`

**Comment by mistermendez** [comment]
Use the following asset to test `Live` and `Video Quality` buttons:

```
pcode: x2aDkyOt2q3WtOCp
playerBrandingId: 378512f55dd44fc287d0fa609353c268
Embed Code: BsZjdieDrJdl9YjLBm1LVyl3FF0KdWtd
```

File: js/components/controlBar.js (hunk 9 above)
Comment (shruthi17) [inline js/components/controlBar.js]: ok, working on this.

**Comment by mistermendez** [comment]
👍

**Pull request merged**